kolibri/core/assets/src/views/CoreMenu/CoreMenuOption.vue

```diff
@@ -83,12 +130,56 @@
       optionIconColor() {
         return this.active ? this.$themeTokens.primary : null;
       },
+      iconAfter() {
+        return this.visibleSubMenu ? 'chevronUp' : 'chevronDown';
+      },
+    },
+    mounted() {
+      this.submenuShouldBeOpen();
     },
     methods: {
-      conditionalEmit() {
-        if (!this.link) {
-          this.$emit('select');
+      isActiveLink(route) {
+        // temporary second condition to account for downloads route structure
+        if (!this.$route.name || this.$route.name != 'MY_DOWNLOADS') {
+          return route.includes(this.$route.path);
+        }
+      },
+      submenuShouldBeOpen() {
+        if (this.subRoutes && this.subRoutes.length > 0) {
+          // temporary second condition to account for downloads route structure
+          window.location.pathname.includes(this.link) && this.$route.name != 'MY_DOWNLOADS'
+            ? (this.visibleSubMenu = true)
+            : (this.visibleSubMenu = false);
         }
+        return false;
+      },
+      subpathStyles(route) {
+        if (this.isActiveLink(route) && this.$router.currentRoute.name != 'PROFILE') {
```

Comment:
Why is Profile special?

If it is special, then this should be specified as a property in the Profile nav item, and then conditionalized on that.